test/unit/RelayServer.test.ts

```diff
@@ -1,688 +1,535 @@
-            const methodStub = {
-                call: stub().returns(true)
-            };
-            const maxPossibleGasWithViewCall =
-                await server.findMaxPossibleGasWithViewCall(
-                    methodStub,
-                    req,
-                    maxPossibleGas
-                );
-            expect(maxPossibleGasWithViewCall.toString()).to.be.equal(
-                maxPossibleGas
-            );
-        });
-
-        it('should return an accepted max possible gas after applying factor', async function () {
-            const methodCallStub = stub();
-            methodCallStub.onCall(0).throws(Error('Not enough gas left'));
-            methodCallStub.onCall(1).returns(true);
-            const methodStub = {
-                call: methodCallStub
-            };
-            const maxPossibleGasWithViewCall =
-                await server.findMaxPossibleGasWithViewCall(
-                    methodStub,
-                    req,
-                    maxPossibleGas
-                );
-            const expectedGasLimit =
-                BigNumber(1.25).multipliedBy(maxPossibleGas);
-            expect(maxPossibleGasWithViewCall.toString()).to.be.equal(
-                expectedGasLimit.toString()
-            );
-        });
-
-        it('should fail if the gas limit is exceeded', async function () {
-            const error = Error('Not enough gas left');
-            const methodStub = {
-                call: stub().throws(error)
-            };
-            const maxPossibleGasWithViewCall =
-                server.findMaxPossibleGasWithViewCall(
-                    methodStub,
-                    req,
-                    maxPossibleGas
-                );
-            await expect(maxPossibleGasWithViewCall).to.be.rejectedWith(
-                GAS_LIMIT_EXCEEDED
-            );
-        });
-
-        it('should fail with a different error', async function () {
-            const error = Error('Fake error');
-            const methodStub = {
-                call: stub().throws(error)
-            };
-            const maxPossibleGasWithViewCall =
-                server.findMaxPossibleGasWithViewCall(
-                    methodStub,
-                    req,
-                    maxPossibleGas
-                );
-            await expect(maxPossibleGasWithViewCall).to.be.rejectedWith(
-                error.message
-            );
-        });
+              request: {
+                tokenContract: constants.AddressZero,
+                tokenAmount: lowTokenAmount,
+              },
+              relayData: {
+                gasPrice: GAS_PRICE,
+              },
+            },
+          } as EnvelopingTxRequest)
+        ).to.be.rejectedWith(
+          'User agreed to spend lower than what the transaction may require'
+        );
+      });
+    });
+  });
+
+  describe('Comparisons between estimateMaxPossibleGas() and getMaxPossibleGas()', function () {
```

Comment:
Although I understand the reason behind this method, I was wondering if we can manage to have a single method used in both places, so we wouldn't need to compare them.

Reply:
We have calculateFee() and both methods call it. Even though I think the reason to have this tests is to make sure that those two methods return a similar value regardless of the internal implementation.